shivyc/il_cmds/math.py

```diff
@@ -118,6 +118,76 @@ class Mult(_AddMult):
     Inst = asm_cmds.Imul
 
 
+class _BitShiftCmd(ILCommand):
+    """Base class for bitwise shift commands."""
+
+    # The ASM instruction to generate for this command. Override this value
+    # in subclasses.
+    Inst = None
+
+    def __init__(self, output, arg1, arg2): # noqa D102
+        self.output = output
+        self.arg1 = arg1
+        self.arg2 = arg2
+
+    def inputs(self): # noqa D102
+        return [self.arg1, self.arg2]
+
+    def outputs(self): # noqa D102
+        return [self.output]
+
+    def rel_spot_pref(self): # noqa D102
+        return {self.output: [self.arg1, self.arg2]}
```

Comment:
This `rel_spot_pref` function tells the register allocator to attempt to allocate `self.arg1` or `self.arg2` in the same spot (i.e. register) as `self.output` if possible. But you don't actually want `self.arg2` in this list because there's no benefit to putting `self.arg2` and `self.output` in the same register. (But if `self.arg1` and `self.output` were in the same register, that would save us a `mov` operation!)